pkg/repo/mysql/services_integration_test.go

```diff
@@ -0,0 +1,254 @@
+	return &model.Service{
+		Name: gofakeit.FirstName(),
+		ID:   uuid.New(),
+	}
+}
+
+func Test_SaveService(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	tests := []struct {
+		service *model.Service
+		name    string
+	}{
+		{
+			name:    "With UUID",
+			service: randService(),
+		},
+		{
+			name:    "Without UUID",
+			service: &model.Service{Name: gofakeit.Name()},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			err := repository.SaveService(ctx, tt.service)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			// check if really saved
+			actualService, err := repository.LoadService(ctx, tt.service.ID)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			assert.Equal(t, tt.service, actualService)
+		})
+	}
+}
+
+func Test_UpdateService(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	expectedService := randService()
+
+	err := repository.SaveService(ctx, expectedService)
+	if !assert.NoError(t, err, "Prepare test data") {
+		return
+	}
+
+	// update service fields and save
+	expectedService.Name = gofakeit.FirstName()
+
+	err = repository.SaveService(ctx, expectedService)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// check if really updated
+	actualService, err := repository.LoadService(ctx, expectedService.ID)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	assert.Equal(t, expectedService, actualService)
+}
+
+func Test_LoadService(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	service := randService()
+
+	err := repository.SaveService(ctx, service)
+	if !assert.NoError(t, err, "Prepare test data") {
+		return
+	}
+
+	tests := []struct {
+		expected    *model.Service
+		expectedErr error
+		name        string
+		serviceID   uuid.UUID
+	}{
+		{
+			name:        "All OK",
+			serviceID:   service.ID,
+			expected:    service,
+			expectedErr: nil,
+		},
+		{
+			name:        "Nonexistent",
+			serviceID:   uuid.New(),
+			expectedErr: repo.ErrNotFound,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			actual, err := repository.LoadService(ctx, tt.serviceID)
+
+			if tt.expectedErr != nil {
+				assert.ErrorContains(t, err, tt.expectedErr.Error())
+				return
+			}
+
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			assert.Equal(t, tt.expected, actual)
+		})
+	}
+}
+
+func Test_LoadServices(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	expectedServices := make([]model.Service, 3)
+
+	for i := 0; i < 3; i++ {
+		service := randService()
+
+		err := repository.SaveService(ctx, service)
+		if !assert.NoError(t, err, "Prepare test data") {
+			return
+		}
+
+		expectedServices[i] = *service
+	}
+
+	actual, err := repository.LoadServices(ctx)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	for _, expected := range expectedServices {
+		if !assert.Contains(t, actual, expected) {
+			return
+		}
+	}
```

Comment:
```suggestion
	assert.ElementsMatch(t, expected, actual)
```

Reply:
`ElementsMatch` asserts that `list1` contains the same elements as `list2`. So this assertion needs a length of lists to be equal. In this case, the `actual` variable returned by `ListServices()` holds all the services inserted into the database. Since their sizes are not equal (3 vs. n, where n is the number of records in the database), this assertion will not work.